schema/signature.go

```diff
@@ -0,0 +1,24 @@
+package schema
+
+import (
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+type FuncSignature struct {
+	// Description is an optional human-readable description
+	// of the function.
+	Description string
+
+	// ReturnType is the ctyjson representation of the function's
+	// return types based on supplying all parameters using
+	// dynamic types. Functions can have dynamic return types.
+	ReturnType cty.Type
+
+	// Parameters describes the function's fixed positional parameters.
+	Params []function.Parameter
+
+	// VariadicParameter describes the function's variadic
+	// parameter if it is supported.
+	VarParam *function.Parameter
```

Comment:
```suggestion
	// Params describes the function's fixed positional parameters.
	Params []function.Parameter

	// VarParam describes the function's variadic
	// parameter if it is supported.
	VarParam *function.Parameter
```

I'm not sure if the short names are better here, but I guess they align with the relevant function names in go-cty, so at least we're consistent 🤷🏻 

 - https://pkg.go.dev/github.com/zclconf/go-cty@v1.12.1/cty/function#Function.Params
 - https://pkg.go.dev/github.com/zclconf/go-cty@v1.12.1/cty/function#Function.VarParam